services/server/src/server/services/workers/verificationWorker.ts

```diff
@@ -256,6 +267,219 @@ async function _verifyFromEtherscan({
   });
 }
 
+function createPreRunCompilationFromCandidate(
+  candidate: GetSourcifyMatchByChainAddressWithPropertiesResult,
+): PreRunCompilation | null {
+  if (
+    !candidate.std_json_input ||
+    !candidate.std_json_output ||
+    !candidate.version ||
+    !candidate.fully_qualified_name
+  ) {
+    logger.debug("Incomplete similarity candidate data", {
+      chainId: candidate.chain_id,
+      address: candidate.address,
+    });
+    return null;
+  }
+  const language = candidate.std_json_input.language;
+  const { contractName, contractPath } = splitFullyQualifiedName(
+    candidate.fully_qualified_name,
+  );
+
+  const compilationTarget = {
+    name: contractName,
+    path: contractPath,
+  };
+
+  try {
+    if (language === "Solidity") {
+      return new PreRunCompilation(
+        solc,
+        candidate.version,
+        candidate.std_json_input,
+        candidate.std_json_output,
+        compilationTarget,
+        candidate.creation_cbor_auxdata || {},
+        candidate.runtime_cbor_auxdata || {},
+      );
+    } else if (language === "Vyper") {
+      const compilation = new PreRunCompilation(
+        vyper,
+        candidate.version,
+        candidate.std_json_input,
+        candidate.std_json_output,
+        compilationTarget,
+        candidate.creation_cbor_auxdata || {},
+        candidate.runtime_cbor_auxdata || {},
+      );
+      if (candidate.metadata) {
+        compilation.setMetadata(candidate.metadata);
+      }
+      return compilation;
+    }
+  } catch (error: any) {
+    logger.debug("Failed to create PreRunCompilation for candidate", {
+      chainId: candidate.chain_id,
+      address: candidate.address,
+      language,
+      error: error?.message,
+    });
+    return null;
+  }
+
+  logger.debug("Unsupported language for similarity candidate", {
+    chainId: candidate.chain_id,
+    address: candidate.address,
+    language,
+  });
+  return null;
+}
+
+async function _verifySimilarity({
+  chainId,
+  address,
+  runtimeBytecode,
+  creatorTxHash,
+  candidates,
+}: VerifySimilarityInput): Promise<VerifyOutput> {
+  const sourcifyChain = chainRepository.sourcifyChainMap[chainId];
+  if (!sourcifyChain) {
+    logger.warn("Similarity verification requested for unsupported chain", {
+      chainId,
+      address,
+    });
+    return {
+      errorExport: {
+        customCode: "internal_error",
+        errorId: uuidv4(),
+      },
+    };
+  }
```

Comment:
In the other functions we just expect the `sourcifyChain` to be present. If you want to check for it, maybe do it in all functions of the worker. Also, I think we should just throw and not return an error export. The verification service will handle this case and already set it to `internal_error`.